File: src/java/PhantasyIslandJavaRemoteControl/radio/DebugRadio.java
package PhantasyIslandJavaRemoteControl.radio;

import java.util.List;

/**
 * Debug 示例代码，对应 Python 的 debug_radio.py。
 */
public class DebugRadio {
    public static void main(String[] args) {
        RadioManager rm = new RadioManager();
        rm.connect();

        System.out.println(rm.createMsgTimestampId());
        System.out.println(rm.ping());

        // sync 模式（默认）
        Object result = rm.radioApi.listRadioLocalObjectsIds();
        List<String> ids = ApiModule.asSync(result);
        System.out.println(ids);

        System.out.println("isSceneInit: " + rm.radioApi.isSceneInit());
        System.out.println("isRadioReachabilityCheckerInit: " + rm.radioApi.isRadioReachabilityCheckerInit());
        System.out.println("getAllRadioMaterial: " + rm.radioApi.getAllRadioMaterial());
        System.out.println("localRadioMaterial: " + rm.radioApi.localRadioMaterial());
        System.out.println("listRadioLocalObjectsIds: " + rm.radioApi.listRadioLocalObjectsIds());

        // rm.waitExit(); // 假设有类似的方法
    }
}

File: src/kt/PhantasyIslandKotlinRemoteControl/PhantasyIslandKotlinRemoteControl/radio/DebugRadio.kt
package PhantasyIslandKotlinRemoteControl.radio

/**
 * 调试 RadioManager 的简单示例程序
 */
fun main() {
    val rm = RadioManager()
    rm.connect()

    println("Timestamp ID: ${rm.createMsgTimestampId()}")

    // 默认 SYNC 模式
    val pingResult = rm.debugApi.ping()
    println("Ping Result: $pingResult")

    val localObjectIds = asSync<List<String>>(rm.radioApi.listRadioLocalObjectsIds())
    println("Local Object IDs: $localObjectIds")

    println("isSceneInit: ${rm.radioApi.isSceneInit()}")
    println("isRadioReachabilityCheckerInit: ${rm.radioApi.isRadioReachabilityCheckerInit()}")
    println("getAllRadioMaterial: ${rm.radioApi.getAllRadioMaterial()}")
    println("localRadioMaterial: ${rm.radioApi.localRadioMaterial()}")

    // 可以在这里添加更多测试逻辑
    
    // 如果需要保持程序运行以等待异步回调
    // Thread.sleep(10000)
}
